src/uct/ib/base/ib_md.c

```diff
@@ -1835,7 +1835,8 @@ ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
                           UCT_MD_FLAG_NEED_RKEY |
                           UCT_MD_FLAG_ADVISE;
 
-    if (md->config.odp.max_size == UCS_MEMUNITS_AUTO) {
+    if ((md->config.odp.max_size == UCS_MEMUNITS_AUTO) &&
+        !(md->cap_flags & UCT_MD_FLAG_REG_ODP)) {
```

Comment:
can we remove md->config.odp.max_size altogether?
it's a reminiscent from older devices.
we can say ODP does not have size limit when supported. so if umr_extended_translation_offset is 0, ODP is not supported.